EHR_SM/src/org/labkey/ehr_sm/EHR_SMCustomizer.java

```diff
@@ -0,0 +1,204 @@
+    {
+        SQLFragment ageSql = null;
+        if (demographics.getSqlDialect().isSqlServer())
+        {
+            ageSql = new SQLFragment("(SELECT CONVERT(DECIMAL(10,2), DATEDIFF(month, dem.birth, ").append(receivedCol.getValueSql(alias));
+            ageSql.append(") / 12.0) as ageAtSample FROM ").append(demographics, "dem");
+            ageSql.append(" WHERE dem.ParticipantId = ").append(idCol.getValueSql(alias)).append(")");
+        }
+        else if (demographics.getSqlDialect().isPostgreSQL())
+        {
+            ageSql = new SQLFragment("(SELECT ROUND(CAST((EXTRACT(YEAR FROM age) + EXTRACT(MONTH FROM age) / 12) AS NUMERIC), 2) AS ageAtSample ");
+            ageSql.append("FROM age(").append(receivedCol.getValueSql(alias)).append(", ");
+            ageSql.append("(SELECT dem.birth FROM ").append(demographics, "dem");
+            ageSql.append(" WHERE dem.ParticipantId = ").append(idCol.getValueSql(alias)).append(")) AS t(age))");
+        }
+
+        return ageSql;
+    }
+
+    private void customizeSamples(AbstractTableInfo ti)
+    {
+        MutableColumnInfo inputsCol = (MutableColumnInfo) ti.getColumn("Inputs");
+        String ehrContainerName = getEHRContainer(ti.getUserSchema().getContainer());
+        if (null != inputsCol && null != ehrContainerName)
+        {
+            // Allows fks on the Animal source and provides message if more than a single Animal source in lineage
+            ((ExpLineageForeignKey) inputsCol.getFk()).setSingleValueForeignKey("Animal");
+
+            // Id column info must be queried after setSingleValueForeignKey to ensure correct lookups
+            FieldKey idFk = FieldKey.fromParts("Inputs", "Data", "Animal", "Id");
+            ColumnInfo idCol = QueryService.get().getColumns(ti, Arrays.asList(idFk)).get(idFk);
+
+            if (null != idCol)
+            {
+                // Add to default columns
+                List<FieldKey> defaultCols = new ArrayList<>(ti.getDefaultVisibleColumns());
+
+                FieldKey animalRecordFk = FieldKey.fromParts("AnimalRecord");
+                var aliasCol = new AliasedColumn(idCol.getParentTable(), animalRecordFk, idCol, false);
+                aliasCol.setShownInInsertView(false);
+                aliasCol.setShownInUpdateView(false);
+
+                aliasCol.setDisplayColumnFactory(new DisplayColumnFactory()
+                {
+                    @Override
+                    public DisplayColumn createRenderer(ColumnInfo colInfo)
+                    {
+                        return new DataColumn(colInfo)
+                        {
+
+                            @Override
+                            public Object getValue(RenderContext ctx)
+                            {
+                                String value = (String) super.getValue(ctx);
+                                if (value != null && value.startsWith("#Error"))
```

Comment:
If we stick with handling multiple values with this string constant, can we use a constant somewhere?